### PR TITLE
[MO] Align MO namespaces

### PR DESCRIPTION
**Pull request by pavel-esir**
In order to align Model Optimizer with the new namespace convention
- `model-optimizer/...` was renamed & moved -> `tools/mo/...`
- `mo/` and `extensions/` directories were merged into `mo/.` and moved from `model-optimizer/` into `tools/mo/openvino/tools/mo`
- all imports were changed `mo` -> `openvino.tools.mo, extensions` -> `openvino.tools.mo`
- accordingly changed `setup.py` to deploy MO into `setup-packages/openvino/tools/mo`
- updated imports in `POT`
- updated CMakeLists.txt from `tools/mo/`
- updated docs
 
**Validation:**
* [x] check if CI infrastructure fits the new MO directory structure: successfully passed
* [x] check that offline_transformations are correctly loaded: done on Windows and Linux, left to check of Mac
* [x] check that `python setup.py install` deploys correct folder structure
* [x] check that changes do not break `POT`: checked manually
 
**Docs:**
* [x]  update Model Optimizer conversion docs

**Nit (can be addressed during bugfix):**
* [ ] check if precompiled *.egg with protobuf for Caffe is still needed

**Tickets:**
 - CVS-62457, CVS-60366

### CODE REVIEW

File: model-optimizer/setup.py
Comment (slyubimt) [inline model-optimizer/setup.py:67]: Is it done for backward compatibility? I believe we do not need this workaround with remapping modules in __init__.py created from setup.py script. Instead of that python modules should be correctly allocated inside  /openvino/tools/mo directory structure. @dkurt, please to comment.

**Comment by pavel-esir** [comment]
> Maybe it is load diff issue, but seems like pkgutil-style __init__.py was not provided in this PR https://packaging.python.org/guides/packaging-namespace-packages/#pkgutil-style-namespace-packages . Without it import of other OpenVINO modules (tools installed along with model-optimizer) from `openvino` namespace will be broken.

Added pkgutil-style __init__.py files

**Comment by KodiaqQ** [comment]
Are there any changes for POT? POT contains imports from mo at least. Should they be aligned with the imports from openvino.tools.mo?

**Comment by pavel-esir** [comment]
> Are there any changes for POT? POT contains imports from mo at least. Should they be aligned with the imports from openvino.tools.mo?

@KodiaqQ thanks for noting this! Updated POT imports as well https://github.com/openvinotoolkit/openvino/pull/7708/commits/bd0e580818c92d27092aebda8cc5783ecf3bf432

File: docs/HOWTO/Custom_Layers_Guide.md
Comment (pavel-esir) [reply docs/HOWTO/Custom_Layers_Guide.md:140]: For some reason this line has left unchanged. Will update together with subsequent changes

File: docs/HOWTO/Custom_Layers_Guide.md
Comment (pavel-esir) [reply docs/HOWTO/Custom_Layers_Guide.md:261]: For some reason this line has left unchanged. Will update together with subsequent changes

File: docs/HOWTO/Custom_Layers_Guide.md
Comment (pavel-esir) [inline docs/HOWTO/Custom_Layers_Guide.md:331]: revert to mri_reconstruction_demo.py

Comment (pavel-esir) [reply docs/HOWTO/Custom_Layers_Guide.md:331]: fixed

File: docs/HOWTO/Custom_Layers_Guide.md
Comment (pavel-esir) [reply docs/HOWTO/Custom_Layers_Guide.md]: done

File: docs/HOWTO/Custom_Layers_Guide.md
Comment (pavel-esir) [reply docs/HOWTO/Custom_Layers_Guide.md]: done

**Review by KodiaqQ — APPROVED** [review]

**Comment by KodiaqQ** [comment]
> > Are there any changes for POT? POT contains imports from mo at least. Should they be aligned with the imports from openvino.tools.mo?
> 
> @KodiaqQ thanks for noting this! Updated POT imports as well [bd0e580](https://github.com/openvinotoolkit/openvino/commit/bd0e580818c92d27092aebda8cc5783ecf3bf432)

@pavel-esir, can you please add the try-except handle for new and old MO imports to make ci/pot green and handle both cases?

File: scripts/setupvars/setupvars.sh
Comment (slyubimt) [inline scripts/setupvars/setupvars.sh:76]: These lines are not relevant any more and should be remove, since we remove mo from installer

Comment (pavel-esir) [reply scripts/setupvars/setupvars.sh:76]: @slyubimt thanks for noting that! I will remove them

Comment (pavel-esir) [reply scripts/setupvars/setupvars.sh]: removed

**Review by slyubimt — APPROVED** [review]

**Comment by azhogov** [comment]
CODEOWNERS should be changed as well

**Review by azhogov — APPROVED** [review]

**Comment by pavel-esir** [comment]
> CODEOWNERS should be changed as well

Fortunately CODEOWNERS is already updated https://github.com/pavel-esir/openvino/blob/align-mo-namespaces/CODEOWNERS#L66

File: docs/MO_DG/prepare_model/convert_model/Converting_Model_General.md
Comment (rkazants) [inline docs/MO_DG/prepare_model/convert_model/Converting_Model_General.md:144]: Please remove "File can be specified as relative path 
                        from the current directory, as absolute path or as a 
                        relative path from the mo root directory"
I don't think it is super necessary thing that we are proud of.

Comment (rkazants) [inline docs/MO_DG/prepare_model/convert_model/Converting_Model_General.md:144]: I don't think we need it "File can be specified as relative path 
                        from the current directory, as absolute path or as a 
                        relative path from the mo root directory"
Please remove it. It looks evident

File: docs/MO_DG/prepare_model/convert_model/pytorch_specific/Convert_F3Net.md
Comment (rkazants) [inline docs/MO_DG/prepare_model/convert_model/pytorch_specific/Convert_F3Net.md:31]: ./mo -> mo

File: docs/MO_DG/prepare_model/convert_model/pytorch_specific/Convert_QuartzNet.md
Comment (rkazants) [inline docs/MO_DG/prepare_model/convert_model/pytorch_specific/Convert_QuartzNet.md:32]: ./mo -> mo

File: docs/MO_DG/prepare_model/convert_model/pytorch_specific/Convert_RCAN.md
Comment (rkazants) [inline docs/MO_DG/prepare_model/convert_model/pytorch_specific/Convert_RCAN.md:30]: ./mo -> mo

File: docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_Slim_Library_Models.md
Comment (rkazants) [inline docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_Slim_Library_Models.md:44]: please use prefix for directory with installed site-packages <>/mo

File: docs/MO_DG/prepare_model/convert_model/Converting_Model_General.md
Comment (pavel-esir) [reply docs/MO_DG/prepare_model/convert_model/Converting_Model_General.md]: removed this lines

File: docs/MO_DG/prepare_model/convert_model/pytorch_specific/Convert_F3Net.md
Comment (pavel-esir) [reply docs/MO_DG/prepare_model/convert_model/pytorch_specific/Convert_F3Net.md]: done

File: docs/MO_DG/prepare_model/convert_model/pytorch_specific/Convert_RCAN.md
Comment (pavel-esir) [reply docs/MO_DG/prepare_model/convert_model/pytorch_specific/Convert_RCAN.md]: done

File: docs/MO_DG/prepare_model/convert_model/pytorch_specific/Convert_QuartzNet.md
Comment (pavel-esir) [reply docs/MO_DG/prepare_model/convert_model/pytorch_specific/Convert_QuartzNet.md]: done

File: docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_Slim_Library_Models.md
Comment (pavel-esir) [reply docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_Slim_Library_Models.md]: added `<PYTHON_SITE_PACKAGES>` prefix

File: docs/get_started/get_started_windows.md
Comment (rkazants) [inline docs/get_started/get_started_windows.md:260]: mo

File: docs/how_tos/MonoDepth_how_to.md
Comment (rkazants) [inline docs/how_tos/MonoDepth_how_to.md:6]: please revert the change

File: docs/how_tos/MonoDepth_how_to.md
Comment (rkazants) [inline docs/how_tos/MonoDepth_how_to.md:64]: please do not change it. I believe it should be done separately in the scope of other task

File: tools/mo/openvino/tools/mo/analysis/tf_retinanet.py
Comment (rkazants) [inline tools/mo/openvino/tools/mo/analysis/tf_retinanet.py:62]: please check a path. It seems incorrect

File: tools/mo/openvino/tools/mo/analysis/tf_yolo.py
Comment (rkazants) [inline tools/mo/openvino/tools/mo/analysis/tf_yolo.py:75]: please correct a path

File: tools/mo/openvino/tools/mo/analysis/tf_yolo.py
Comment (rkazants) [inline tools/mo/openvino/tools/mo/analysis/tf_yolo.py:102]: please correct a path

File: tools/mo/openvino/tools/mo/back/GatherNormalizer.py
Comment (rkazants) [inline tools/mo/openvino/tools/mo/back/GatherNormalizer.py:1]: why is this marked as added?

File: tools/mo/openvino/tools/mo/back/StridedSliceMasksNormalizer.py
Comment (rkazants) [inline tools/mo/openvino/tools/mo/back/StridedSliceMasksNormalizer.py:1]: why is this marked as added?

File: tools/mo/openvino/tools/mo/back/ie_ir_ver_2/emitter.py
Comment (rkazants) [inline tools/mo/openvino/tools/mo/back/ie_ir_ver_2/emitter.py:279]: please correct the path

File: tools/mo/openvino/tools/mo/back/pass_separator.py
Comment (rkazants) [inline tools/mo/openvino/tools/mo/back/pass_separator.py:1]: marked as added

Comment (pavel-esir) [reply tools/mo/openvino/tools/mo/back/pass_separator.py:1]: It's a bug of github display. Indeed it's moving, I did it via `git mv`.

After merge git blame and history will be kept with no issues (i double checked that).

File: tools/mo/openvino/tools/mo/back/GatherNormalizer.py
Comment (pavel-esir) [reply tools/mo/openvino/tools/mo/back/GatherNormalizer.py:1]: the reason is explained below

**Comment by pavel-esir** [comment]
@rkazants addressed comments

**Review by rkazants — APPROVED** [review]
conditionally approved if custom CI run will be fully green

**Comment by azhogov** [comment]
CI passed: https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/360

**Pull request merged**
